### ISSUE TITLE
bug(datepicker): It is announcing the date buttons type incorrectly as text elements

### ISSUE THREAD

**Issue by zarend** [angular/components#23476]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Go to the "Basic datepicker" demo
3. Activate "Open calendar" button from any section, say, "Single date selection"
4. Focus on any date value
 
#### Expected Behavior
It should announce the date values as buttons/links so that screen readers can understand them to be clickable

#### Actual Behavior

It is announcing the date values as text elements using VoiceOver

#### Environment
OS| Browser | Screen reader
macOS Catalina | Google Chrome 92.0.4515.131 | VoiceOver

##### Additional Notes
This is copied from an [internal bug report](http://b/195600292) found during a11y testing.

**Comment by zarend** [comment]
@jelbourn , from what I can tell, Voiceover does not announce that `gridcell`'s are intractable when they have `aria-selected="true"` or `aria-selected="false"`. To back that up, [a11ysupport.io's page on aria-select gridcell](https://a11ysupport.io/tests/aria_gridcell(aria-selected)) has a minimal example of this. 

If we were to work around that, putting a button inside each `td` seems like the most reasonable approach to me.
